src/components/application_manager/include/application_manager/application_manager_impl.h

```diff
@@ -415,6 +415,14 @@ class ApplicationManagerImpl
    * @brief Closes all registered applications
    */
   void UnregisterAllApplications();
+
+  void AddAppToActivate(ApplicationConstSharedPtr app,
```

Comment:
@LitvinenkoIra generally, it is a good idea to keep tracking of apps with postponed activation requests, but it is too much to have a new container for that as for me.
I propose to replace the container with simple property of `ApplicationImpl` itself:
```
std::atomic<uint32_t> pending_activation_corr_id_; // set to 0 in ctor

bool IsActivationPending() const {
  return pending_activation_corr_id_ > 0;
}

void SwapActivationPendingCorrId(uint32_t& corr_id) {
  std::swap(pending_activation_corr_id_, corr_id);
}
```

Then you may remove `IsAppWaitingForActivation` and `AddAppToActivate` function but keep `ActivatePostponedApp`: 
```
 SDL_LOG_AUTO_TRACE();
 uint32_t corr_id = 0;
 app->Swap(corr_id); 
 policy_handler_->OnActivateApp(app->app_id(), corr_id);
```

Please update all other pieces of code accordingly

Reply:
@AKalinich-Luxoft thanks for the review. One point related to the suggsted approach - we have the "app" as a constant object in the SDLActivateAppRequest https://github.com/LuxoftSDL/sdl_core/pull/146/files#diff-eb07db0c8f5a6918ce619f1adac00f7a49671d3939f3bbf71564f4a1d85c8795R154 
I believe this was done in order not to change the app while the request is being processed. If we decide to add the pending_activation_corr_id_ property, we will need to change the application while processing the SDLActivateApp request. Are you okay with such changes?

Reply:
@LitvinenkoIra okay, lets keep const object then.
In that case, please follow your initial approach but make a few changes there:
1. Move out all new methods from AM into the separate component like `PostponedActivationController`
2. Make this component as a field in `StateController` class and make it public accessible
3. Inside of `PostponedActivationController` keep track of app_id instead of holding of shared pointer (may cause memory leaks)
4. Don't forget to remove application record when application unregisters
5. Use that component in all other places

Reply:
@AKalinich-Luxoft Done in a2c4436eacfc0bc4eaef5b434c3e04b6c3595caa